### PR TITLE
Migrate the gke-prod e2e test from the 1.1 release branch to the 1.2 release branch

### PR DESCRIPTION
**Pull request by roberthbailey**
And add a gke-prod-parallel test.

### CODE REVIEW

**Comment by spxtr** [comment]
LGTM

One nit is that we were assuming jobs would be parallel by default, and any serial jobs would be suffixed with `-serial`. I'll probably want to rename these to fit that convention as a part of #23021.

**Comment by roberthbailey** [comment]
Roger. I was following the naming convention you put in place recently for staging. :)

**Comment by roberthbailey** [comment]
Rebased.

File: hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml [hunk 1]
```diff
@@ -390,18 +408,7 @@
     suffix:
         - 'gke-1.1':
             timeout: 150
-            description: 'Run E2E tests on GKE from the current release branch.'
-            cron-string: 'H */6 * * *'
```
Comment (spxtr) [inline hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml]: I think we want to keep this line. We're running 1.1 tests less frequently now.

Comment (roberthbailey) [inline hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml]: Good catch (I missed that while resolving the merge conflict).

**Comment by roberthbailey** [comment]
PTAL.

**Pull request merged**